tools/autograd/derivatives.yaml

```diff
@@ -871,6 +871,12 @@
 - name: lu_unpack(Tensor LU_data, Tensor LU_pivots, bool unpack_data=True, bool unpack_pivots=True) -> (Tensor P, Tensor L, Tensor U)
   LU_data: lu_unpack_backward(grads, LU_data, unpack_data)
   LU_pivots: non_differentiable
+  # regarding P. It does not change, hence its differential is zero.
+  # It is possible to mark it as such with setting output_differentiability: [False, True, True],
+  # but it causes OpInfo test_dtypes to fail.
```

Comment:
@albanD , could you tell whether it is expected? A better approach is to skip defining dP at all.

Reply:
That is not expected no.
What is the error that you get if you mark it non-differentiable exactly?

Reply:
If I set output_differentiability, I get:
```python
Traceback (most recent call last):
  File "/home/nik/git/Quansight/pytorch/test/test_ops.py", line 173, in test_dtypes
    self.assertEqual(supported_backward_dtypes, claimed_backward_supported, msg=msg)
  File "/home/nik/git/Quansight/pytorch/torch/testing/_internal/common_utils.py", line 1607, in assertEqual
    super().assertEqual(x, y, msg=self._get_assert_msg(msg, debug_msg=debug_msg))
  File "/home/nik/.conda/envs/pytorch-cuda-dev/lib/python3.6/unittest/case.py", line 829, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/home/nik/.conda/envs/pytorch-cuda-dev/lib/python3.6/unittest/case.py", line 1082, in assertSetEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/home/nik/.conda/envs/pytorch-cuda-dev/lib/python3.6/unittest/case.py", line 670, in fail
    raise self.failureException(msg)
AssertionError: Items in the second set but not the first:
torch.complex128
torch.float64
torch.float32
torch.complex64 : Attempted to compare [set] types: Expected: set(); Actual: {torch.complex128, torch.float64, torch.float32, torch.complex64}.
The supported backward dtypes for lu_unpack on cuda according to its OpInfo are
        {torch.complex128, torch.float64, torch.float32, torch.complex64}, but the detected supported backward dtypes are set().
        The following backward dtypes should be removed from the OpInfo: {torch.complex128, torch.float64, torch.float32, torch.complex64}.
============================================================================================= short test summary info =============================================================================================
SKIPPED [1] test/test_ops.py:58: onlyOnCPUAndCUDA: doesn't run on meta
FAILED test/test_ops.py::TestCommonCPU::test_dtypes_lu_unpack_cpu - AssertionError: Items in the second set but not the first:
FAILED test/test_ops.py::TestCommonCUDA::test_dtypes_lu_unpack_cuda - AssertionError: Items in the second set but not the first:

```
I suspect it could be because of the very first output being analyzed for supporting dtypes in differentiability. CC @mruberry

Reply:
Ho the opinfo is actually hiding the actual error. We should save the error messages to explain why each dtype was not supported to improve that error message.

Can you run that test outside of opinfo to see if the error message makes sense?

Reply:
Could you please clarify? If I run a sequence of tests from opinfo, I do not get this long message regarding the supported dtypes, only if I run test_dtypes explicitly. It could be that only the first output in the tuple of returned values is checked for differentiability. And if it is marked as non-differentiable, there is a conflict in OpInfo, as it mentions checking grads for all floating/complex types, but the output produced is non-differentiable.

Reply:
Ho the other gradcheck tests work fine?

My understanding of how that test work is that it just catches all the errors and produce that result.
You can add a `print(e)` herre: https://github.com/pytorch/pytorch/blob/85222c050f3fe2e2c7180f6f0aed5d0bbc7ab671/test/test_ops.py#L128 to see what caused each dtype to fail?

Reply:
But indeed it picks the first Tensor when there are multiple outputs. We should update that code to take the first Tensor that requires grad here: https://github.com/pytorch/pytorch/blob/85222c050f3fe2e2c7180f6f0aed5d0bbc7ab671/test/test_ops.py#L115